language/tools/test_generation/src/abstract_state.rs

```diff
@@ -52,6 +52,18 @@ impl AbstractValue {
         );
         AbstractValue { token, kind }
     }
+
+    /// Create a new struct `AbstractValue` given its type and kind
+    pub fn new_struct(token: SignatureToken, kind: Kind) -> AbstractValue {
+        assert!(
```

Comment:
this should be a precondition!

Reply:
Changed to `checked_precondition!` in upcoming PR.